Sources/TuistDependencies/Carthage/Utils/CarthageGraphGenerator.swift

```diff
@@ -27,18 +27,22 @@ public final class CarthageGraphGenerator: CarthageGraphGenerating {
             .flatMap { $0.allProducts }
 
         let thirdPartyDependencies: [String: ThirdPartyDependency] = Dictionary(grouping: products, by: \.name)
-            .compactMapValues { product in
-                guard let frameworkName = product.first?.container else { return nil }
-
-                let path = AbsolutePath("/")
-                    .appending(components: [
-                        Constants.tuistDirectoryName,
-                        Constants.DependenciesDirectory.name,
-                        Constants.DependenciesDirectory.carthageDirectoryName,
-                        frameworkName,
-                    ])
-
-                return .xcframework(path: path)
+            .compactMapValues { products in
+                guard let product = products.first else { return nil }
+
+                if let xcFrameworkName = product.container {
+                    let path = AbsolutePath("/")
+                        .appending(components: [
+                            Constants.tuistDirectoryName,
+                            Constants.DependenciesDirectory.name,
+                            Constants.DependenciesDirectory.carthageDirectoryName,
+                            xcFrameworkName,
+                        ])
+
+                    return .xcframework(path: path)
+                }
```

Comment:
> Assume that we have multiple platform in Dependencies.swift and we're importing carthage dependencies as binaries. Now it will just hide the framework dependency. I will add support for framework dependencies in graph just a bit later in separate PR if you don't mind

We could print a warning message to console when `product.container` equal to `nil`. The message should inform that product will not added to the graph.

Reply:
✅